collector/generatorreceiver/internal/topology/service_tier.go

```diff
@@ -41,9 +42,37 @@ func (st *ServiceTier) GetResourceAttributeSet() *ResourceAttributeSet {
 }
 
 func (st *ServiceTier) GetRoute(routeName string) *ServiceRoute {
-	for _, v := range st.Routes {
-		if v.Route == routeName {
-			return &v
+	return st.Routes[routeName]
+}
+
+func (st *ServiceTier) Validate(topology Topology) error {
+	for _, m := range st.Metrics {
+		err := m.validate()
+		if err != nil {
+			return fmt.Errorf("error with metric %s in service %s: %v", m.Name, st.ServiceName, err)
+		}
+	}
```

Comment:
I'm sorry I didn't think of this before 😬 but while we're here we should probably validate the flags on `ResourceAttributeSet` as well. But! This also makes for a nice refactor - you can move the `validate` method off of `Metric`, `TagSet` and `ServiceRoute` and onto `EmbeddedFlags` itself (and rename it to `Validate` to export it) and then we won't need to triplicate (or quadruplicate!) the logic.

Reply:
No problem! Yup that makes sense, much nicer